### ISSUE TITLE
Socket connecting infinitely. + https not connecting

### ISSUE THREAD

**Issue by faiziakbr** [infitio/flutter_socket_io#3]
I have implemented the adhara socket.
i have successfully emitted and received the message from server side but the problem is the socket is again connecting and sending the request indefinitely. 
how to stop this?

**Comment by tiholic** [comment]
@faiziakbr can you share the code?

**Comment by faiziakbr** [comment]
Here is the code.
SocketIO socket = await SocketIOManager().createInstance('http://URL:8000/');
    socket.onConnect((data){
      print("connected...");
      print(data);
      socket.emit("upDownVoteRequest", ['Hello World']);
    });
    socket.on("upDownVoteResponse", (data){   //sample event
      print("news");
      print(data);
    });
    socket.connect();

**Comment by mickadam87** [comment]
Hello, I'am sorry but I have approximately the same problem. here my console log and my code attached:

Syncing files to device Android SDK built for x86...
Restarted application in 3 585ms.
I/flutter (30107): initSockets.......
D/Adhara:Socket(30107): Connecting to... http://localhost:3000/
D/Adhara:Socket(30107): registering::connect
D/Adhara:Socket(30107): registering::top_events
D/Adhara:Socket(30107): registering::disconnect
D/Adhara:Socket(30107): registering::reconnect


![capture d ecran 2019-02-01 a 14 50 09](https://user-images.githubusercontent.com/47243665/52126826-b6487b80-2630-11e9-8fb0-8c64272749da.png)

Can you help me please...the app stay blocked on "connecting..."

**Comment by tiholic** [comment]
@mickadam87 I'm working on a fix. Will update soon.

**Comment by mickadam87** [comment]
I have found the problem. I have to use "http://10.0.2.2:3000" in place of "localhost:3000" in Android Virtual Device ! Socket connected ! Thank you so much !

**Comment by mickadam87** [comment]
Sorry, but I have a new issue during Xcode Building...

PATHto MYapp/ios/Pods/Socket.IO-Client-Swift/Source/SocketIO/Engine/SocketEngine.swift:316:11: error: value of type 'WebSocket' has no member 'onHttpResponseHeaders'

Known issue ?

**Comment by tiholic** [comment]
@mickadam87 yes.

https://github.com/infitio/flutter_socket_io/issues/8

**Comment by choe1991** [comment]
I'am sorry but I have approximately the same problem

`D/Adhara:Socket(19495): Socket triggered::reconnect
I/flutter (19495): reconnected to socket.io
D/Adhara:Socket(19495): Socket triggered::connect
D/Adhara:Socket(19495): emitting:::null:::to:::GroupJoin
I/flutter (19495): connected...
D/Adhara:Socket(19495): Socket triggered::disconnect
I/flutter (19495): disconnected from socket.io
D/Adhara:Socket(19495): Socket triggered::reconnect
I/flutter (19495): reconnected to socket.io
D/Adhara:Socket(19495): Socket triggered::connect
I/flutter (19495): connected...
D/Adhara:Socket(19495): emitting:::null:::to:::GroupJoin
D/Adhara:Socket(19495): Socket triggered::disconnect
I/flutter (19495): disconnected from socket.io
D/Adhara:Socket(19495): Socket triggered::reconnect
I/flutter (19495): reconnected to socket.io
D/Adhara:Socket(19495): Socket triggered::connect
I/flutter (19495): connected...
D/Adhara:Socket(19495): emitting:::null:::to:::GroupJoin
D/Adhara:Socket(19495): Socket triggered::disconnect
I/flutter (19495): disconnected from socket.io
D/Adhara:Socket(19495): Socket triggered::reconnect
I/flutter (19495): reconnected to socket.io
D/Adhara:Socket(19495): Socket triggered::connect
D/Adhara:Socket(19495): emitting:::null:::to:::GroupJoin
I/flutter (19495): connected...
`

**Comment by choe1991** [comment]
i have try to clone the example project and changed the url to my own host.the problem still happend.

**Comment by majoferenc** [comment]
For me in the latest version it is working flawlessly on Android side, but iOS is connecting infinitely...

**Comment by theromie** [comment]
I am facing same issue due to this message screen is getting refreshed and load content again

**Comment by mucan54** [comment]
In socket.io this problem happened to me when the server's socket.io version and client's socket.io version was different.

**Comment by agnoam** [comment]
Hey, what is the status of this problem ?
Are you still working on this ?

**Comment by Jasper-zzq** [comment]
I encountered the same problem. I don’t see it when I visit the local.

**Comment by saarang1995** [comment]
@tiholic Hi, have you fixed the issue?
In my current scenario, I am able to connect to the socketIO server which we have on staging but not the one on production using iOS. Let me know if you require any inputs from my end like code snippets.

**Comment by oncleguigs** [comment]
Same issue here, did someone found a fix ?

**Comment by saarang1995** [comment]
After too many trials and failures, I have finally written a solution in Javascript until we have a working solution for Flutter. I am writing a blog about it and will share the solution there with a complete explanation. I will add the link here for reference as soon as possible.

**Comment by saarang1995** [comment]
@oncleguigs @Jasper-zzq @agnoam 
I have come up with a solution using javascript. Please read my blog for the complete solution:
https://saarangtiwari.com/blog/how-i-used-javascript-to-bridge-and-connect-to-websockets-in-flutter

**Comment by SankhaJay** [comment]
hey guys
try adding inside the android/app/src/main/AndroidManifest.xml file this
<application
      ....
      ....
      android:usesCleartextTraffic="true">

**Comment by Atuldubey98** [comment]
> Hello, I'am sorry but I have approximately the same problem. here my console log and my code attached:
> 
> Syncing files to device Android SDK built for x86...
> Restarted application in 4 785ms.
> I/flutter (30107): initSockets.......
> D/Adhara:Socket(30107): Connecting to... http://localhost:3000/
> D/Adhara:Socket(30107): registering::connect
> D/Adhara:Socket(30107): registering::top_events
> D/Adhara:Socket(30107): registering::disconnect
> D/Adhara:Socket(30107): registering::reconnect
> D/Adhara:Socket(30107): Connecting....
> 
> ![capture d ecran 2019-02-01 a 14 50 09](https://user-images.githubusercontent.com/47243665/52126826-b6487b80-2630-11e9-8fb0-8c64272749da.png)
> 
> Can you help me please...the app stay blocked on "connecting..."

Was this problem solved?? Connecting of socket infinite no of times and then disconnecting.

**Comment by mannnish** [comment]
it's 2021 and i am still facing the exact same issue.

**Comment by yusufnadar** [comment]
Any solution?

**Comment by tiholic** [comment]
Please try using new release: V1.0.0 and create a new issue if any problem still exists.

> Note that the this package currently supports only socket.io v2. Track status for v3/v4 socket here: #160 

@yusufnadar @mannnish @Atuldubey98 @SankhaJay @saarang1995 @oncleguigs @Jasper-zzq @agnoam @mucan54 @theromie @majoferenc @choecode @mickadam87 @faiziakbr
